### PR TITLE
Missing PHP 8 Attributes for some annotations

### PR DESCRIPTION
**Pull request by devmaslov**
Within recent updates GraphQLight supports native PHP 8 Attributes - which is amazing!

I stumbled upon a couple annotations which cannot be used as attributes yet, even though there're examples in the docs using the new approach. The ones in question are: `HideParameter`, `InjectUser` and `UseInputType`. Added support for them.

### CODE REVIEW

File: src/Annotations/InjectUser.php [hunk 1]
```diff
@@ -18,6 +19,7 @@
  *   @Attribute("for", type = "string")
  * })
  */
+#[Attribute(Attribute::TARGET_METHOD)]
```
Comment (moufmouf) [inline src/Annotations/InjectUser.php:22]:
```suggestion
#[Attribute(Attribute::TARGET_PARAMETER)]
```

File: src/Annotations/InjectUser.php [hunk 2]
```diff
@@ -26,13 +28,13 @@ class InjectUser implements ParameterAnnotationInterface
     /**
      * @param array<string, mixed> $values
      */
-    public function __construct(array $values = [])
+    public function __construct(array $values = [], ?string $for = null)
```
Comment (moufmouf) [inline src/Annotations/InjectUser.php:31]: You can revert the changes in the constructor. The "for" parameter is only useful with Doctrine annotations. With PHP8 annotations, we will put the annotation on the parameter itself, which is way cleaner.

See https://graphqlite.thecodingmachine.io/docs/argument-resolving#annotations-parsing

**Review by moufmouf — CHANGES_REQUESTED** [review]
Geez! Thanks a lot! I totally failed to migrate those 3 annotations, thanks for pointing that. And thanks for the PR!

We need to change a few things and this should work great.

Annotations extending `ParameterAnnotationInterface` are "special", because they target a parameter. With PHP 8, we can now put attributes directly on parameters (rather than on methods with a "for" attribute).

So all we need to do is actually to put this attribute at the top of the class:

```php
#[Attribute(Attribute::TARGET_PARAMETER)]
```

**Comment by devmaslov** [comment]
@moufmouf, that's a nice catch! I like the idea of using those on parameters directly - indeed it's a cleaner way. I've changed the attribute target to `TARGET_PARAMETER` for all 3 annotations. And made them not repeatable - since it does not make much sense anymore.

File: src/Annotations/InjectUser.php (hunk 1 above)
Comment (devmaslov) [reply src/Annotations/InjectUser.php]: Fixed for all 3 annotations.

File: src/Annotations/InjectUser.php (hunk 2 above)
Comment (devmaslov) [reply src/Annotations/InjectUser.php]: Makes sense! Pushed the adjustments.

**Comment by moufmouf** [comment]
Excellent! Thanks a lot!
